### PR TITLE
PLAT-6243: Add aws backup vault for efs

### PR DESCRIPTION
**Pull request by Secretions**
Based this on Steve's branch with the KMS changes to avoid moving targets

### CODE REVIEW

File: submodules/storage/variables.tf [hunk 1]
```diff
@@ -56,3 +56,39 @@ variable "efs_kms_key" {
   type        = string
   default     = null
 }
+
+variable "efs_backup_vault_kms_key" {
```
Comment (fraenkel) [inline submodules/storage/variables.tf:60]: is this the id or arn?

File: submodules/storage/variables.tf [hunk 2]
```diff
@@ -56,3 +56,39 @@ variable "efs_kms_key" {
...
+variable "efs_backup_vault_force_destroy" {
+  description = "Toggle to allow automatic destruction of all backups when destroying."
+  type        = bool
+  default     = false
+}
+
+variable "efs_backup_schedule" {
+  type        = string
+  description = "Cron-style schedule for EFS backup vault"
+  default     = "0 12 * * ? *"
```
Comment (fraenkel) [inline submodules/storage/variables.tf:81]: state what the default is in the description

**Review by fraenkel — APPROVED** [review]

File: submodules/storage/variables.tf (hunk 1 above)
Comment (Secretions) [reply submodules/storage/variables.tf:60]: It's the arn. This is keeping the pattern with the other variables in Steve's PR, so I'll leave any discussion on the variable name for there (and make sure it matches here at the time of merging).

File: submodules/storage/variables.tf (hunk 2 above)
Comment (fraenkel) [reply submodules/storage/variables.tf:81]: I meant in English since most folks wont understand cron syntax

**Pull request merged**